### ISSUE TITLE
truncation string is not working reliablely with NSLineBreakByTruncatingTail

### ISSUE THREAD

**Issue by Yue-Wang-Google** [facebookarchive/AsyncDisplayKit#2140]
Test case: 

Apply the following patch to the Kittens example:

```
--- /Users/wangyue/Downloads/AsyncDisplayKit-master/examples/Kittens/Sample/KittenNode.mm   2016-08-25 10:03:06.000000000 -0700
+++ Sample/KittenNode.mm    2016-08-25 13:00:22.000000000 -0700
@@ -98,8 +98,12 @@

   // lorem ipsum text, plus some nice styling
   _textNode = [[ASTextNode alloc] init];
+  _textNode.truncationMode = NSLineBreakByTruncatingTail;
+  _textNode.maximumNumberOfLines = 2;
   _textNode.attributedString = [[NSAttributedString alloc] initWithString:[self kittyIpsum]
                                                                attributes:[self textStyle]];
+  _textNode.truncationAttributedText = [[NSAttributedString alloc] initWithString:@"\u2026 More"
+                                                                       attributes:[self textStyle]];
   [self addSubnode:_textNode];

   // hairline cell separator
```

See the first node  in the following screenshot. 

![screen shot 2016-08-25 at 1 04 44 pm](https://cloud.githubusercontent.com/assets/20689178/17984049/8ae77e7a-6ac4-11e6-8eec-dd5b11771363.png)

**Comment by Yue-Wang-Google** [comment]
OK it seems ASDK calculate the truncation point either too early or too late. When using NSLineBreakByTruncatingTail, seems the line truncates too late. When using without, it happens too early. In a few cases two or three more words can be crammed to the last line, but ASDK just truncate a lot earlier.

**Comment by abbasmousavi** [comment]
I have the same problem, It seems the line truncates too late, When using ```NSLineBreakByTruncatingTail``` and truncates too early when using ```NSLineBreakByWordWrapping```.

Also when ```NSLineBreakByTruncatingTail```  breaks the line too late, it displays default truncate message instead of ```truncationAttributedText``` or ``` additionalTruncationMessage```

**Comment by abbasmousavi** [comment]
@appleguy Could you please take a look at this issue?

**Comment by jdee** [comment]
I'm also using .ByTruncatingTail. I'm adding an NSLinkAttributeName to the truncationAttributedText in order to get taps in textNodeTappedTruncationToken() in my delegate. It works perfectly if there are also linked URLs in the attributedText (with NSLinkAttributeNames). The token is consistently absent if there are no linked URLs in the attributedText. It just trails off with an ellipsis, not the specified truncationAttributedText, like the first cell in the image above.

**Comment by maicki** [comment]
Related: #1284

**Comment by garrettmoon** [comment]
@Adlai-Holler can you take a stab at figuring out if we can fix this in the 2.0 milestone?

**Comment by garrettmoon** [comment]
This issue was moved to TextureGroup/Texture#142
